src/main/java/org/candlepin/subscriptions/capacity/CapacityReconciliationController.java

```diff
@@ -128,11 +132,17 @@ public void enqueueReconcileCapacityForOffering(String sku) {
   }
 
   private void reconcileSubscriptionCapacities(
-      Collection<SubscriptionCapacity> newCapacities, String subscriptionId, String sku) {
+      Collection<SubscriptionCapacity> newCapacities,
+      String ownerId,
+      String subscriptionId,
+      String sku) {
 
     Collection<SubscriptionCapacity> toSave = new ArrayList<>();
     Map<SubscriptionCapacityKey, SubscriptionCapacity> existingCapacityMap =
-        subscriptionCapacityRepository.findByKeySubscriptionId(subscriptionId).stream()
+        subscriptionCapacityRepository
+            .findByKeyOwnerIdAndKeySubscriptionIdIn(
+                ownerId, Collections.singletonList(subscriptionId))
```

Comment:
Switching off of findByKeySubscriptionId and on to findByKeyOwnerIdAndKeySubscriptionIdIn is what provides the majority of our speedup, as it makes use of the compound primary key (owner_id, product_id, subscription_id). 

Performance testing on a particular org improved time-to-sync-100-subs from 39.9 seconds to 2.9 seconds, an over 10x speed up.

For that same test org, Performance tests showed time-to-sync the entire org went from 36m 47s to 2m 52s, again an over 10x speed up.